source/MRMesh/MRPolylineComponents.cpp

```diff
@@ -75,6 +104,47 @@ UnionFind<MR::UndirectedEdgeId> getUnionFindStructure( const PolylineTopology& t
     return unionFindStructure;
 }
 
+template <typename V>
+UndirectedEdgeBitSet getLargestComponent( const Polyline<V>& polyline )
+{
+    MR_TIMER;
+
+    auto& topology = polyline.topology;
+    auto unionFindStruct = getUnionFindStructure( topology );
+    auto region = undirectedEdges( topology );
+
+    const auto& allRoots = unionFindStruct.roots();
+    auto [uniqueRootsMap, k] = getUniqueRoots( allRoots, region );
+
+    auto maxLength = std::numeric_limits<float>::lowest();
+    int maxI = 0;
+    std::vector<float> areas( k, 0.f );
+    for ( auto e : region )
+    {
+        auto index = uniqueRootsMap[allRoots[e]];
+        auto& area = areas[index];
+        area += polyline.edgeLength( EdgeId( e ) );
+        if ( area > maxLength )
+        {
+            maxI = index;
+            maxLength = area;
+        }
+    }
+
+    UndirectedEdgeBitSet maxLengthComponent( allRoots.size() );
```

Comment:
better `polyline.topology.lastNonLoneEdge() + 1` instead of `allRoots.size()`

Reply:
Fixed.